src/components/header/Header.tsx

```diff
@@ -0,0 +1,159 @@
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+import SearchBox from "../searchbox/SearchBox";
+import ProfileMenu from "../../pages/homepage/ProfileMenu";
+import { useNavigate } from "react-router-dom"; // Import useNavigate
+import { toast } from "sonner";
+import styled from "styled-components";
+
+export default function Header(props: { onSearchChange: (event: React.ChangeEvent<HTMLInputElement>) => void }) {
+
+  const navigate = useNavigate(); // For navigating after logout
+
+  // Check for user token or username in local storage
+  const isLoggedIn = localStorage.getItem("token");
```

Comment:
Local storage calls could be combined. It's more a cosmetic change though.
```typescript
const { token: isLoggedIn, username } = {
    token: localStorage.getItem("token"),
    username: localStorage.getItem("username")
  };
```

Reply:
OK, I adjusted it